negotiate.go

```diff
@@ -5,7 +5,7 @@ import "net/http"
 var processors = []ResponseProcessor{&JSONProcessor{}, &XMLProcessor{}}
 
 //New sets up response processors. By default XML and JSON are created
-func New(responseProcessors []ResponseProcessor) {
+func New(responseProcessors ...ResponseProcessor) {
```

Comment:
shouldn't this also be `func New(repsonseProcessors ...*RepsonseProcessor) {` so we pass around pointers properly, or do we want to copy things for the processors?